### ISSUE TITLE
Quickdoc virtualtext doesn't disappear when I expect it to

### ISSUE THREAD

**Issue by daveyarwood** [Olical/conjure#46]
### Overview

The conjure settings in my vimrc include:

```viml
let g:conjure_quick_doc_normal_mode = 0
```

I like seeing quickdoc virtualtext when I'm in insert mode, e.g. if I start typing `(when `, I see the docstring for `clojure.core/when` as expected.

When I go back into normal mode, I expect the quickdoc virtualtext to disappear, but it doesn't; instead, it sticks around until the next time I enter insert mode and start typing a form.

### Screenshot

![2019-07-01-164420_1244x188_scrot](https://user-images.githubusercontent.com/4909079/60465599-7efdeb00-9c1f-11e9-8d2d-42d7d2693530.png)

**Comment by Olical** [comment]
Thanks for reporting! I'm aware of this one and have it on my trello of small tweaks I want to make, I'll try to expedite the change now that someone else has thought of it too. Also I removed the logs from the issue, we won't be needing them for this and it's pretty long.

It's not a bug as such, just something that I realised will have to happen in certain configurations. I don't want to clear it every time you change mode, so maybe I need to clear it on mode change depending on your settings. The logic just gets a little tangled.
